src/python/WMCore/MicroService/Unified/MSRuleCleaner.py

```diff
@@ -50,27 +116,341 @@ def execute(self, reqStatus):
+        """
+        msg = "Called with wflow: %s"
+        self.logger.debug(msg, pformat(wflow))
+        # NOTE: The following dispatch logic is a subject to be changed at any time
+
+        # Do not clean any Resubmission, but still let them be archived
+        if wflow['RequestType'] == 'Resubmission':
+            wflow['ForceArchive'] = True
+            msg = "Skipping cleanup step for workflow: %s - RequestType is %s."
+            msg += "\nWill try to archive it directly."
+            self.logger.info(msg, wflow['RequestName'], wflow['RequestType'])
+        # Clean:
+        elif wflow['RequestStatus'] in ['rejected', 'aborted-completed']:
+            # NOTE: We do not check the ParentageResolved flag for these
+            #       workflows, but we do need to clean output data placement
+            #       rules from the agents for them
+            for pline in self.agentlines:
+                try:
+                    pline.run(wflow)
+                except Exception as ex:
+                    msg = "%s: General error from pipeline. Error: \n%s. "
+                    msg += "\nWill retry again in the next cycle."
+                    self.logger.exception(msg, pline.name, str(ex))
+                    continue
+                if wflow['CleanupStatus'][pline.name]:
+                    self.wfCounters['cleaned'][pline.name] += 1
+        elif wflow['RequestStatus'] == 'announced' and not wflow['ParentageResolved']:
+            # NOTE: We skip workflows which are not having 'ParentageResolved'
+            #       flag, but we still need some proper logging for these.
+            msg = "Skipping workflow: %s - 'ParentageResolved' flag set to false."
+            msg += "\nWill retry again in the next cycle."
+            self.logger.info(msg, wflow['RequestName'])
+        elif wflow['RequestStatus'] == 'announced' and wflow['ParentageResolved']:
+            for pline in self.cleanuplines:
+                try:
+                    pline.run(wflow)
+                except Exception as ex:
+                    msg = "%s: General error from pipeline. Error: \n%s. "
+                    msg += "\nWill retry again in the next cycle."
+                    self.logger.exception(msg, pline.name, str(ex))
+                    continue
+                if wflow['CleanupStatus'][pline.name]:
+                    self.wfCounters['cleaned'][pline.name] += 1
+        else:
+            # We shouldn't be here:
+            msg = "Skipping workflow: %s - "
+            msg += "Does not fall under any of the defined categories."
+            self.logger.info(msg, wflow['RequestName'])
+
+        # Archive:
+        try:
+            self.plineArchive.run(wflow)
+            self.wfCounters['archived'] += 1
+        except MSRuleCleanerArchival as ex:
+            msg = "%s: Archival Error: %s. "
+            msg += "\nWill retry again in the next cycle."
+            self.logger.error(msg, wflow['PlineMarkers'][-1], ex.message())
```

Comment:
Same comment, should we add the workflow name in the error message and write it all in the same line? When using the logger object, breaking these in multiple lines make things look bad (besides losing the timestamp from a possible grep command).

Reply:
The workflow name is already in the message coming from the `archive()` method, which puts it directly into the exception message. See the log messages related to the accumulated action history of a non-archived workflow [1]. 
The Redundant new line was removed.

Done

[1]
```
2020-11-06 15:51:40,397:DEBUG:MSRuleCleaner:cleanRucioRules(): plineAgentCont: 'TransferStatus' is 'pending' or 'TransferInfo' is missing. Skip deleting any rules for: tivanov_TaskChain_Prod_HG2011_Val_201029_112740_7125 
2020-11-06 15:51:40,427:DEBUG:MSRuleCleaner:cleanRucioRules(): plineAgentBlock: 'TransferStatus' is 'pending' or 'TransferInfo' is missing. Skip deleting any rules for: tivanov_TaskChain_Prod_HG2011_Val_201029_112740_7125 
2020-11-06 15:51:40,449:ERROR:MSRuleCleaner:_dispatchWflow(): plineArchive: Archival Error: MSRuleCleanerArchival: Not properly cleaned workflow: tivanov_TaskChain_Prod_HG2011_Val_201029_112740_7125. Will retry again in the next cycle.
```